File: app/src/main/java/com/example/phonebook/viewModels/AuthorisationActivityViewModel.java
package com.example.phonebook.viewModels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;

import com.example.phonebook.dagger.ComponentProvider;
import com.example.phonebook.model.repos.ServiceRepoShPref;

import javax.inject.Inject;

public class AuthorisationActivityViewModel extends AndroidViewModel {

    @Inject
    ServiceRepoShPref mServiceRepo;

    public AuthorisationActivityViewModel(@NonNull Application application) {
        super(application);
        ComponentProvider.getInstance().addActivityComponent().inject(this);
    }

    public boolean init() {
        return mServiceRepo.getSignedUp();
    }

}

File: app/src/main/java/com/example/phonebook/viewModels/AuthorisationActivityViewModel.kt
package com.example.phonebook.viewModels

import androidx.lifecycle.ViewModel
import com.example.phonebook.dagger.scopes.ActivitiesScope
import com.example.phonebook.model.repos.ServiceRepoShPref
import javax.inject.Inject

@ActivitiesScope
class AuthorisationActivityViewModel
@Inject constructor(private var mServiceRepo: ServiceRepoShPref) : ViewModel() {
    fun init() = mServiceRepo.getSignedUp()
}